app/models/bookmark.rb

```diff
@@ -27,6 +28,16 @@ def self.auto_delete_preferences
   validate :bookmark_limit_not_reached
   validates :name, length: { maximum: 100 }
 
+  before_create do
+    if self.for_topic
+      self.bookmarkable_type = "Topic"
+      self.bookmarkable_id = Post.find(self.post_id).topic_id
```

Comment:
`Post.find` may return an error if the post has been deleted, do we need to handle the error should that happen?

Reply:
These checks are already done in `BookmarkManager`. This is the only place bookmarks are created apart from Fabricators:

https://github.com/discourse/discourse/blob/62f393b933902f4b101cd9b081f7654e602b5401/lib/bookmark_manager.rb#L36-L50

I put the before_create inside Bookmark to handle Fabricators mostly. Perhaps I will just update the Fabricators and move this to the `BookmarkManager` instead.

Reply:
Moved to trigger.